modules/externalipam/ipam_api.rb

```diff
@@ -0,0 +1,389 @@
+        halt 500, { error: ERRORS[:no_connection] }.to_json
+      end
+    end
+
+    # Get a list of groups from External IPAM
+    #
+    # Returns:
+    #   Response if success:
+    #   ===========================
+    #     Http Code:     200
+    #     JSON Response:
+    #       {"data": [
+    #         {"id": "1", "name": "Group 1", "description": "This is group 1"},
+    #         {"id": "2", "name": "Group 2", "description": "This is group 2"}
+    #       ]}
+    #
+    #   Response if no groups exist:
+    #   ===========================
+    #     Http Code:     404
+    #     JSON Response:
+    #       {"error": "Groups are not supported"}
+    #
+    #   Response if groups are not supported:
+    #   ===========================
+    #     Http Code:     500
+    #     JSON Response:
+    #       {"error": "Groups are not supported"}
+    get '/groups' do
+      content_type :json
+
+      begin
+        halt 500, { error: ERRORS[:groups_not_supported] }.to_json unless provider.groups_supported?
+        groups = provider.ipam_groups
+        groups.to_json
+      rescue Proxy::Validations::Error => e
+        logger.warn(e.message)
+        halt 400, { error: e.to_s }.to_json
+      rescue RuntimeError => e
+        logger.warn(e.message)
+        halt 500, { error: e.message }.to_json
+      rescue Errno::ECONNREFUSED, Errno::ECONNRESET
+        logger.warn(ERRORS[:no_connection])
+        halt 500, { error: ERRORS[:no_connection] }.to_json
+      end
+    end
+
+    # Get a group from External IPAM
+    #
+    # Inputs:   1. group:       The name of the External IPAM group
+    #
+    # Returns:
+    #   Response if success:
+    #   ===========================
+    #     Http Code:     200
+    #     JSON Response:
+    #       {"data": {"id": "1", "name": "Group 1", "description": "This is group 1"}}
+    #
+    #   Response if group not found:
+    #   ===========================
+    #     Http Code:     404
+    #     JSON Response:
+    #       {"error": "Group not Found"}
+    #
+    #   Response if groups are not supported:
+    #   ===========================
+    #     Http Code:     500
+    #     JSON Response:
+    #       {"error": "Groups are not supported"}
+    get '/groups/:group' do
+      content_type :json
+
+      begin
+        validate_required_params!([:group], params)
+
+        group_name = get_request_group(params)
+        group = provider.get_ipam_group(group_name)
+
+        halt 404, { error: ERRORS[:no_group] }.to_json if group.nil?
+        group.to_json
+      rescue Proxy::Validations::Error => e
+        logger.warn(e.message)
+        halt 400, { error: e.to_s }.to_json
+      rescue RuntimeError => e
+        logger.warn(e.message)
+        halt 500, { error: e.message }.to_json
+      rescue Errno::ECONNREFUSED, Errno::ECONNRESET
+        logger.warn(ERRORS[:no_connection])
+        halt 500, { error: ERRORS[:no_connection] }.to_json
+      end
+    end
```

Comment:
Now that we're on Ruby 2.5 we can use `rescue` at the method level.
```suggestion
      validate_required_params!([:group], params)

      group_name = get_request_group(params)
      group = provider.get_ipam_group(group_name)

      halt 404, { error: ERRORS[:no_group] }.to_json if group.nil?
      group.to_json
    rescue Proxy::Validations::Error => e
      logger.warn(e.message)
      halt 400, { error: e.to_s }.to_json
    rescue RuntimeError => e
      logger.warn(e.message)
      halt 500, { error: e.message }.to_json
    rescue Errno::ECONNREFUSED, Errno::ECONNRESET
      logger.warn(ERRORS[:no_connection])
      halt 500, { error: ERRORS[:no_connection] }.to_json
    end
  end
```